cmd/erasure-multipart.go

```diff
@@ -432,13 +443,17 @@ func (er erasureObjects) PutObjectPart(ctx context.Context, bucket, object, uplo
 	partsMetadata, errs = readAllFileInfo(rctx, storageDisks, minioMetaMultipartBucket,
 		uploadIDPath, "", false)
 
+	// Unlock upload id locks before, so others can get it.
+	uploadIDRLock.RUnlock(rlkctx.Cancel)
```

Comment:
I think it is wrong to unlock after reading partsMetadata, because what happens next is that we update partsMetadata with checksum info and save partsMetadata again in the disks, however a parallel part upload could update in paralllel and we can lose some information in xl.meta

Reply:
@vadmeste We write lock + reload it before we write anything.

Reply:
> @vadmeste We write lock + reload it before we write anything.

Yes I missed it.